python/tskit/trees.py

```diff
@@ -2498,14 +2498,35 @@ def map_mutations(self, genotypes, alleles):
         :param tuple(str) alleles: The alleles for the specified ``genotypes``. Each
             positive value in the ``genotypes`` array is treated as an index into this
             list of alleles.
+        :param ancestral_state: A fixed ancestral state, specified either as a
+            non-negative integer less than the number of alleles, or a string which
+            must be one of the ``alleles`` provided above. If ``None`` (default) then
+            the most parsimonious ancestral state is chosen. Note that if the ancestral
+            state is specified, mutations can occur above the root node(s) of the tree,
+            for example if all the samples have a genotype of 1 but the ancestral state
+            is fixed to be 0.
+        :type ancestral_state: Union[int, str]
         :return: The inferred ancestral state and list of mutations on this tree
             that encode the specified observations.
         :rtype: (str, list(tskit.Mutation))
         """
         genotypes = util.safe_np_int_cast(genotypes, np.int8)
-        if np.max(genotypes) >= 64:
+        max_alleles = np.max(genotypes)
+        if ancestral_state is not None:
+            try:
```

Comment:
I don't see the advantage of this try except over ``isinstance(ancestral_state, str)``, which is clearer in intent

Reply:
Happy to use `isinstance`. ISTR you thought use of is instance was indicative of poor code structure, that's all. I can't see a way around this sort of thing here, though.

Reply:
Can't really avoid isinstance when doing type polymorphism, it's better than opaque try/excepts